ComponentKit/Core/CKComponentMemoizer.mm

```diff
@@ -164,14 +171,17 @@ id CKMemoize(CKMemoizationKey memoizationKey, id (^block)(void))
   return impl ? impl->_next : nil;
 }
 
-CKComponentLayout CKMemoizeOrComputeLayout(CKComponent *component, CKSizeRange constrainedSize, const CKComponentSize& size, CGSize parentSize)
+CKComponentLayout CKMemoizeLayout(CKComponent *component, CKSizeRange constrainedSize, const CKComponentSize& size, CGSize parentSize, CKComponentLayout (^block)())
 {
-  if (component && [component shouldMemoizeLayout]) {
```

Comment:
This checks if the component is non-nil. Is that something we still need to guarantee?